### ISSUE TITLE
ffmpeg decode fails in Cygwin

### ISSUE THREAD

**Issue by duckpuppy** [sampsyo/audioread#27]
I've been having issues on Cygwin with beets and the chroma plugin, but they all seem to be rooted around the audioread package.  The problem is that the filename passed into the popen_multiple command needs to be quoted for Cygwin Python.  I was getting `file not found` errors from beets when I first started trying to run it with the chroma plugin in Cygwin.  Once I edited the `ffdec.py` file to add quotes around the filename those went away and the ffmpeg process began running properly.

**Comment by sampsyo** [comment]
Wow! That would be *tremendously disappointing* if Cygwin is futzing with command arguments even when a shell is not used. Thanks for pointing this out.

I'll investigate a bit more when I have access to a Cygwin setup.

**Comment by sampsyo** [comment]
Weirdly, I haven't able to produce any bad behavior on Cygwin.

```py
>>> import platform
>>> platform.platform()
'CYGWIN_NT-10.0-WOW-1.7.35-0.287-5-3-i686-32bit'
>>> import subprocess
>>> subprocess.Popen([u'cat', u"foo ' \" \xe9"])
<subprocess.Popen object at 0xffdaa4ec>
hello
>>> 
```

Popen was able to correctly invoke a command on a file with a very funky name, with no quotes necessary.

Could you perhaps do a little more digging and find out exactly what kinds of filenames trigger the bad behavior? We might then be able to narrow down the problem.

**Comment by duckpuppy** [comment]
I can't seem to reproduce it myself now.  I have recompiled a new version of ffmpeg in the meantime, and I wonder if somehow it was more ffmpeg than audioread... adding the quotes to the filename argument got me past an IO exception of `file not found` and allowed ffmpeg to attempt to execute.  At the time, ffmpeg would then error out for another reason, which is why I ended up recompiling a new version and now it works.  Removing my hack to add quotes around the filename now also works.  I'm a bit confused as to why that might be, but it's working now.  Python on Cygwin does have some quirks around process management (I know at least of a couple revolving around process ids specifically), so I jumped ahead of myself assuming this was another Cygwin Python process quirk.  Sorry, I will close this myself unless I can get it to happen again.
